tests/ui/traits/non_lifetime_binders/issue-118697.rs

```diff
@@ -0,0 +1,8 @@
+#![allow(incomplete_features)]
+#![feature(non_lifetime_binders)]
+
+type T = dyn for<V = A(&())> Fn(());
```

Comment:
We should probably just ban parameter defaults in non-lifetime binders before we get to AST->HIR lowering.

Reply:
Yes, I agree this is a promising approach. I'll give it a try.